### PR TITLE
Issue 1642: IRP for compromised sys admin acct

### LINKED ISSUE #1642: [Incident Response Drill] As a sys admin, I need to know how to regain control of my compromised account
### Description:

As noted [here](https://github.com/raft-tech/TANF-app/issues/885#issuecomment-1042364716), we will practice how to regain control of a TDP privileged user account that has been compromised and save our incident response plan for this drill in the repo.  This ticket is to track this work. 

### ACs:
- [x] mitigation steps in [hackmd](https://hackmd.io/Yyo_NiIzRp6tqIS-6gLC6w) complete and accurate
- [x] drill conducted
- [x] README is updated, if necessary
- [x] incident response plan (ato version) updated, if necessary

### Tasks:
- [x] finalize drill plan
- [x] conduct drill
- [x] open PR to add drill docs to repo

### PR DESCRIPTION
**Pull request by ADPennington**
## Summary of Changes

- creates new subdir  with [`readme`](https://github.com/raft-tech/TANF-app/tree/069471528f0a8aba9fa69f19a98b883c7281ea0c/docs/Security-Compliance/Incidence-Response) for incident response protocols
- adds incident response [documentation](https://github.com/raft-tech/TANF-app/blob/069471528f0a8aba9fa69f19a98b883c7281ea0c/docs/Security-Compliance/Incidence-Response/Sys-Admin-Acct-Mgmt.md) for restoring sys admin account if it has been compromised
- updates [`readme`](https://github.com/raft-tech/TANF-app/tree/069471528f0a8aba9fa69f19a98b883c7281ea0c/docs/Security-Compliance) for `tanf-app/docs/security-compliance` to reference new subdir. 
- moves [`Secret-Key-Mgmt.md`](https://github.com/raft-tech/TANF-app/blob/069471528f0a8aba9fa69f19a98b883c7281ea0c/docs/Security-Compliance/Incidence-Response/Secret-Key-Mgmt.md) into new IR subdir

Pull request closes #1642

### CODE REVIEW

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# System Admin Account Restoration Strategy
+
+## Scenario: Sys Admin Account(s) Compromised
+
+**TDP has 2 sys admins, and one has turned evil :smiling_imp: and decided to compromise the other's account such that they can no longer access TDP DAC. Below is a screenshot of the compromised account:**
+
+_Sys admin account (i.e. `alexandra.pennington@acf.hhs.gov`)  has been modified with a different username/email address, first name, and last name._
+![](https://i.imgur.com/416KxSy.png)
```
Comment (abottoms-coder) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:8]: You can insert this inline via github so we never lose an outside link to another service.

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+_When the sys admin tries to sign back in, she is taken to the request access page. She will be unable to access TDP DAC until this request is approved by a user with `OFA System Admin` permissions_
+
+![](https://i.imgur.com/urYLj5a.png)
```
Comment (abottoms-coder) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:12]: Again imgur vs github.

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+user.is_superuser = True
+user.deactivated = False
+user.save()
+```
+**4. Login to TDP frontend via ACF AMS and confirm access to TDP DAC restored.**
+
+_Note: be sure to request access again, if prompted. after submitting the request, all appropriate access should be restored._
+
+_TDP frontend view:_
+![](https://i.imgur.com/4hpyAIY.png)
```
Comment (abottoms-coder) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:108]: Again, imgur vs. github

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+- Mitigation steps outlined above target `hhs-acf-prototyping` org, `tdp-staging` space and `tdp-backend-staging` app. <org>, <space>, and <app> targets will need to be modified in a production context. 
+- Review the latest TDP Incidence Response Plan [here](https://hhsgov.sharepoint.com/sites/TANFDataPortalOFA/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTANFDataPortalOFA%2FShared%20Documents%2Fcompliance&viewid=6ecbc5f1%2Dfa9c%2D4b0a%2Da454%2D35e222e8044e), and [see incidence response tips from cloud.gov](https://cloud.gov/docs/ops/security-ir/). 
+
+### Q&A
+- Should Sys owner and sys admins all have the same <org>  roles in [cloudgov](https://docs.cloudfoundry.org/concepts/roles.html#roles)? Is this an acceptable risk? 
+  - Yes. System owner should reach out to cloud.gov support if a privileged user's cloud.gov account roles are compromised.  
+- Compromised account will still appear as a user in TDP DAC. Should we delete this user account? 
+  - No. This account will remained in a deactivated state. 
+- The evil user's account should-- at a minimum-- be deactivated from TDP, but should we delete it (presume this person will no longer be working for ACF)
```
Comment (abottoms-coder) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:148]:
```suggestion
- The evil user's account should-- at a minimum-- be deactivated from TDP, but should we delete it? (presume this person will no longer be working for ACF) 
```

**Review by abottoms-coder — APPROVED** [review]
Made some minor nits/notes but looks good, thanks, Alex!

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+### Communication protocol if `sys admin` account compromised
+- Contact system owner via [Mattermost OFA TDP Product Channel](https://mattermost.goraft.tech/goraft/channels/ofa-tdp-product). If Mattermost isn't available, please send an email and cc: ACF Tech Lead, Product Manager, vendor Tech Lead. 
+
+- Include details of the compromise, which should include:
+    - where compromise occurred (e.g. TDP? cloud.gov?)
+    - how the compromise occurred (if known)
+    - preliminary assessment of the scope and impact of the compromise (e.g. were spaces and/or services deleted? was PII compromised? exposed?)
+    - Next steps (e.g. schedule mtg to discuss incident response plan, rotate keys, etc.)
+    
+- If system admin's compromised account is not restored by following the abovementioned steps, this could because the evil user also changed the admin's cloud.gov account permissions and/or compromised other components of TDP. If this is the case, the `system owner` should reach out to cloud.gov (https://cloud.gov/docs/help/) support to have any necessary user permissions and/or services (e.g. database) restored.
```
[the remark this answers is not in the document]
Comment (stevenino) [reply docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:130]: Just one nit. All else looks good.

**Review by stevenino — APPROVED** [review]

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md (hunk 1 above)
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:8]:
```suggestion
![incident_sysadmin](https://user-images.githubusercontent.com/63075587/162066471-cc5a1f88-aada-4309-9629-b341e61cdb5d.PNG)
```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md (hunk 2 above)
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:12]:
```suggestion
![incident_sysadmin1](https://user-images.githubusercontent.com/63075587/162066631-86c90d4c-1168-48e2-941f-9212c811581e.PNG)

```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+user.is_superuser = True
+user.deactivated = False
+user.save()
+```
+**4. Login to TDP frontend via ACF AMS and confirm access to TDP DAC restored.**
+
+_Note: be sure to request access again, if prompted. after submitting the request, all appropriate access should be restored._
+
+_TDP frontend view:_
+![](https://i.imgur.com/4hpyAIY.png)
```
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:109]:
```suggestion
![incident_sysadmin2](https://user-images.githubusercontent.com/63075587/162066853-6d9b78b3-4425-4881-9fee-8b975a1b7180.PNG)

```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+**4. Login to TDP frontend via ACF AMS and confirm access to TDP DAC restored.**
+
+_Note: be sure to request access again, if prompted. after submitting the request, all appropriate access should be restored._
+
+_TDP frontend view:_
+![incident_sysadmin2](https://user-images.githubusercontent.com/63075587/162066853-6d9b78b3-4425-4881-9fee-8b975a1b7180.PNG)
+
+
+_TDP backend DAC view:_
+![](https://i.imgur.com/PkwjXNU.png)
```
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:113]:
```suggestion
![incident_sysadmin3](https://user-images.githubusercontent.com/63075587/162067096-039cc639-60ed-4c4d-abc5-b2c9f6571bf0.PNG)

```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+### Notes
+
+- Tools needed for this incident response strategy:
+    - cloud.gov account with `user` and `manager` roles assigned at <org> and <space> levels.
+    - [git bash](https://git-scm.com/downloads) - _this usually comes with the download package for Git for Windows. Can alternatively use Command Prompt._
+    - [Curl for Windows 64 bit](https://curl.se/windows/) - _this usually comes with the download package for Git for Windows_
+    - [cloud foundry binary for Windows 64 bit](https://github.com/cloudfoundry/cli/blob/master/doc/installation-instructions/installation-instructions-v7.md#installers-and-compressed-binaries)
+    - [JQuery version 1.6 for Windows 64 bit](https://stedolan.github.io/jq/download/)
+- Mitigation steps outlined above target `hhs-acf-prototyping` org, `tdp-staging` space and `tdp-backend-staging` app. <org>, <space>, and <app> targets will need to be modified in a production context.
```
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:143]:
```suggestion
- Mitigation steps outlined above target `hhs-acf-prototyping` org, `tdp-staging` space and `tdp-backend-staging` app. `org`, `space`, and `app` targets will need to be modified in a production context. 
```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+```
+ssh -p 22 cf:<PROCESS_GUID>/0@ssh.fr.cloud.gov
+```
+* You will be prompted to enter a password. Use the one-time ssh passcode  generated from `cf ssh-code` above (Note: pasting the value will be invisible). If this is successful, the command line will appear like this:
+    
+```
+vcap@SomeHashValue...
+```
+
+* Initiate interactive SSH session using the following 2 commands (press <enter> after each command):
```
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:82]:
```suggestion
* Initiate interactive SSH session using the following 2 commands (press `enter` after each command):
```

File: docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+![incident_sysadmin1](https://user-images.githubusercontent.com/63075587/162066631-86c90d4c-1168-48e2-941f-9212c811581e.PNG)
+
+
+## Example mitigation steps
+- Follow [communication protocol](#communication-protocol-if-sys-admin-account-compromised)
+- Sys admin with compromised account should work with System owner to confirm appropriate admin roles are assigned to their cloud.gov account. This account will be necessary to change TDP DAC access remotely. 
+- Sys admin with compromised account should use CF CLI to remotely restore permissions: 
+
+**<details><summary>CLI commands</summary>**
```
Comment (ADPennington) [inline docs/Security-Compliance/IR/Sys-Admin-Acct-Mgmt.md:20]:
```suggestion
**<details><summary>CF CLI commands</summary>**
```

**Review by lfrohlich — APPROVED** [review]

**Pull request merged**